llvm/lib/Target/AMDGPU/AMDGPUTargetMachine.cpp

```diff
@@ -676,6 +676,11 @@ void AMDGPUTargetMachine::registerPassBuilderCallbacks(
 
         if (EarlyInlineAll && !EnableFunctionCalls)
           PM.addPass(AMDGPUAlwaysInlinePass());
+
+#if __has_feature(address_sanitizer)
```

Comment:
The host compiler cannot matter here.

Reply:
Have made changes to #87265 pass to process only functions which have "sanitize_address" attribute. Will be remove this __has_feature check.